watertap/property_models/unit_specific/activated_sludge/modified_asm2d_properties.py

```diff
@@ -383,68 +486,127 @@ def build(self):
+                / self.params.CODtoVSS_XBM
+                + self.conc_mass_comp["X_PHA"] / self.params.CODtoVSS_XPHA
             )
+            return vss
 
-        self.eq_VSS = pyo.Constraint(rule=rule_VSS)
-
-    def _ISS(self):
-        self.ISS = pyo.Var(
-            initialize=1,
-            domain=pyo.NonNegativeReals,
-            doc="Inorganic suspended solids",
-            units=pyo.units.kg / pyo.units.m**3,
-        )
+        self.VSS = pyo.Expression(rule=_VSS, doc="Volatile suspended solids")
 
-        # TODO: Several HFO and other terms omitted since not included yet.
-        def rule_ISS(b):
-            return (
-                b.ISS
-                == b.params.f_ISS_BM
+        def _ISS(self):
+            iss = (
+                self.params.f_ISS_BM
                 * (
-                    b.conc_mass_comp["X_H"]
-                    + b.conc_mass_comp["X_PAO"]
-                    + b.conc_mass_comp["X_AUT"]
+                    self.conc_mass_comp["X_H"]
+                    + self.conc_mass_comp["X_PAO"]
+                    + self.conc_mass_comp["X_AUT"]
                 )
-                / b.params.CODtoVSS_XBM
-                + b.params.ISS_P * b.conc_mass_comp["X_PP"]
+                / self.params.CODtoVSS_XBM
+                + self.params.ISS_P * self.conc_mass_comp["X_PP"]
             )
+            return iss
+
+        self.ISS = pyo.Expression(rule=_ISS, doc="Inorganic suspended solids")
+
+        def _TSS(self):
+            tss = self.VSS + self.ISS
+            return tss
+
+        self.TSS = pyo.Expression(rule=_TSS, doc="Total suspended solids")
+
+        def _COD(self):
+            cod = (
+                self.conc_mass_comp["S_F"]
+                + self.conc_mass_comp["S_A"]
+                + self.conc_mass_comp["S_I"]
+                + self.conc_mass_comp["X_I"]
+                + self.conc_mass_comp["X_S"]
+                + self.conc_mass_comp["X_H"]
+                + self.conc_mass_comp["X_PAO"]
+                + self.conc_mass_comp["X_PHA"]
+                + self.conc_mass_comp["X_AUT"]
+            )
+            return cod
+
+        self.COD = pyo.Expression(rule=_COD, doc="Chemical oxygen demand")
+
+        def _SNKj(self):
+            snkj = (
+                self.conc_mass_comp["S_NH4"]
+                + self.params.i_NSF * self.conc_mass_comp["S_F"]
+                + self.params.i_NSI * self.conc_mass_comp["S_I"]
+                + self.params.i_NXI * self.conc_mass_comp["X_I"]
+                + self.params.i_NXS * self.conc_mass_comp["X_S"]
+                + self.params.i_NBM
+                * (
+                    self.conc_mass_comp["X_H"]
+                    + self.conc_mass_comp["X_PAO"]
+                    + self.conc_mass_comp["X_AUT"]
+                )
+            )
+            return snkj
 
-        self.eq_ISS = pyo.Constraint(rule=rule_ISS)
+        self.SNKj = pyo.Expression(rule=_SNKj, doc="Kjeldahl nitrogen")
```

Comment:
I am curious why SNKj is short for "Kjeldahl nitrogen", I remember generally we called "total Kjeldahl nitrogen" as TKN

Reply:
The Flores-Alsina files are inconsistent when naming this variable. Sometimes they simply refer to it as Kjeldahl nitrogen and other times they refer to it as total Kjeldahl nitrogen. I opted to use the former, but I agree with you that we should just refer to this as TKN